examples/randlanet_classification.py

```diff
@@ -0,0 +1,255 @@
+        self.mlp_post_attention = SharedMLP([d_out, d_out])
+        self.num_neighbors = num_neighbors
+
+    def forward(self, edge_index, x, pos):
+        out = self.propagate(edge_index, x=x, pos=pos)  # N, d_out
+        out = self.mlp_post_attention(out)  # N, d_out
+        return out
+
+    def message(self, x_j: Tensor, pos_i: Tensor, pos_j: Tensor,
+                index: Tensor) -> Tensor:
+        """Local Spatial Encoding (locSE) and attentive pooling of features.
+
+        Args:
+            x_j (Tensor): neighboors features (K,d)
+            pos_i (Tensor): centroid position (repeated) (K,3)
+            pos_j (Tensor): neighboors positions (K,3)
+            index (Tensor): index of centroid positions
+                (e.g. [0,...,0,1,...,1,...,N,...,N])
+
+        returns:
+            (Tensor): locSE weighted by feature attention scores.
+
+        """
+        # Encode local neighboorhod structural information
+        pos_diff = pos_j - pos_i
+        distance = torch.sqrt((pos_diff * pos_diff).sum(1, keepdim=True))
+        relative_infos = torch.cat([pos_i, pos_j, pos_diff, distance],
+                                   dim=1)  # N * K, d
+        local_spatial_encoding = self.mlp_encoder(relative_infos)  # N * K, d
+        local_features = torch.cat([x_j, local_spatial_encoding],
+                                   dim=1)  # N * K, 2d
+
+        # Attention will weight the different features of x
+        # along the neighborhood dimension.
+        att_features = self.mlp_attention(local_features)  # N * K, d_out
+        att_scores = softmax(att_features, index=index)  # N * K, d_out
+
+        return att_scores * local_features  # N * K, d_out
+
+
+class DilatedResidualBlock(MessagePassing):
+    def __init__(
+        self,
+        num_neighbors,
+        d_in: int,
+        d_out: int,
+    ):
+        super().__init__()
+        self.num_neighbors = num_neighbors
+        self.d_in = d_in
+        self.d_out = d_out
+
+        # MLP on input
+        self.mlp1 = SharedMLP([d_in, d_out // 8])
+        # MLP on input, and the result is summed with the output of mlp2
+        self.shortcut = SharedMLP([d_in, d_out], act=None)
+        # MLP on output
+        self.mlp2 = SharedMLP([d_out // 2, d_out], act=None)
+
+        self.lfa1 = LocalFeatureAggregation(d_out // 4, num_neighbors)
+        self.lfa2 = LocalFeatureAggregation(d_out // 2, num_neighbors)
+
+        self.lrelu = torch.nn.LeakyReLU(**lrelu02_kwargs)
+
+    def forward(self, x, pos, batch):
+        edge_index = knn_graph(pos, self.num_neighbors, batch=batch, loop=True)
+
+        shortcut_of_x = self.shortcut(x)  # N, d_out
+        x = self.mlp1(x)  # N, d_out//8
+        x = self.lfa1(edge_index, x, pos)  # N, d_out//2
+        x = self.lfa2(edge_index, x, pos)  # N, d_out//2
+        x = self.mlp2(x)  # N, d_out
+        x = self.lrelu(x + shortcut_of_x)  # N, d_out
+
+        return x, pos, batch
+
+
+def get_decimation_idx(ptr, decimation):
```

Comment:
Would be great to move this to `nn.pool` IMO.

Reply:
Good idea. I'll make the refactor. :)

Reply:
Done :)